### ISSUE TITLE
Performance issue - Multiple external HTTPS GET request made serving a statuses/:id call

### ISSUE THREAD

**Issue by mhkhung** [mastodon/mastodon#22507]
### Steps to reproduce the problem

Unknown, discovered by monitoring server performance with newrelic.

### Expected behaviour

status#show calls should be served quick 

### Actual behaviour

Making 4 external HTTPS requests 

### Detailed description

I am monitoring my server performance with newrelic and noticed the most time consuming calls are with statuses#show and the traces shows 4 external HTTPS GET requests (with 2 being webfinger requests) to a (small/slow/?) mastodon host that seemingly shouldn't have anything to do with the status requested. What/where is the code doing and why are these calls being made? 

![Screen Shot 2022-12-19 at 11 41 50 AM](https://user-images.githubusercontent.com/1112994/208486695-80fd65dd-99d1-4c2d-9190-f302f24efa46.png)

### Specifications

Mastodon 4.0.2

**Comment by ineffyble** [comment]
That's likely resolving entities referenced directly or indirectly in the status - for example, users who've been mentioned, or users who are mentioned in the pinned post of the author of the status.

**Comment by mhkhung** [comment]
I checked those statuses.. There's no one mentioned, no hashtag etc. I suspect it's something to do with signatures - Trying to pull the public key of the requester to verify signatures?

**Comment by nightpool** [comment]
can you link to the status in question so we can debug this further?

**Comment by ClearlyClaire** [comment]
> I checked those statuses.. There's no one mentioned, no hashtag etc. I suspect it's something to do with signatures - Trying to pull the public key of the requester to verify signatures?

Yes, if the request is signed, it would try fetching the requester's key.

**Comment by mhkhung** [comment]
Here's one of them that I just captured
```
https://mastodon.hongkongers.net/users/RFA_Chinese_mirror/statuses/109548106092489979
```
<img width="1118" alt="Screen Shot 2022-12-20 at 4 40 46 PM" src="https://user-images.githubusercontent.com/1112994/208772275-739a0001-a0cc-424b-a7f2-7726c9185a0f.png">

As @ClearlyClaire said, I am quite sure now it is the code fetching public keys but keeps failing on some hosts.. the stoplight_wrap_request code keeps it from happening more often than every 5 mins or so per host. I think either we increase the `.with_cool_off_time(5.minutes.seconds)` to an hour or so, or somehow be able to flag to the requester something is wrong with their setup..

**Comment by mhkhung** [comment]
I up'ed the logs to Warning and I am surprised I keep seeing it.. not sure why Stoplight's not working as I expected. I emailed that admin and hopefully he's going to fix his server.

```
[aeaf890a-551b-4ea8-aeda-2116052dbc3d] Fetching actor https://mastodon.jg1000.fr/actor failed: Webfinger error when resolving jg1000.fr@mastodon.jg1000.fr: Request for acct:jg1000.fr@jg1000.fr returned HTTP 404
```

**Comment by ClearlyClaire** [comment]
Oh, good catch. We probably want to run the stoplight on more error types, although we need to be careful to report that in a non-confusing way so that implementors or operators do not get misled when trying to debug a federation issue.
